src/input_context/preset.rs

```diff
@@ -0,0 +1,140 @@
+use bevy::prelude::*;
+
+use crate::{Input, InputBind, Negate, SwizzleAxis};
+
+pub trait BindPreset {
+    fn bindings(self) -> impl Iterator<Item = InputBind>;
+}
+
+impl<I: Into<InputBind>> BindPreset for I {
+    fn bindings(self) -> impl Iterator<Item = InputBind> {
+        std::iter::once(self.into())
+    }
+}
+
+/// Maps 4 buttons as 2-dimentional input.
+///
+/// This is a convenience preset that uses [`SwizzleAxis`] and [`Negate`] to
+/// bind the buttons to X and Y axes.
+#[derive(Debug, Clone, Copy)]
+pub struct XyAxis {
+    pub up: Input,
+    pub left: Input,
+    pub down: Input,
+    pub right: Input,
```

Comment:
Small thing to consider.

In my gave I have keyboard settings defined like this:
https://github.com/projectharmonia/project_harmonia/blob/271a425fab59bcf93fabc2a6e3ee8f32940f08be/base/src/settings.rs#L119-L123

They are `Vec`s since you can assign multiple inputs to moving forward, for example.

And because of this I don't use presets in my game, I just assign them manually:
https://github.com/projectharmonia/project_harmonia/blob/271a425fab59bcf93fabc2a6e3ee8f32940f08be/base/src/game_world/player_camera.rs#L228-L243

And I wondering if your preset suggestion could address this. I think we can turn `XyAxis` into `XyAxes` and accept iterators for each direction. This way I could convert keys from settings into the preset. Other presets will be unchanged. What do you think? Is it worth it? Or maybe you have a different idea?

Reply:
I disagree with `with` changing to `with_preset`. `with(KeyCode::W)` feels very natural - `with(GamepadStick::Left)` should feel just as natural. Same for `with(WasdKeys)`.

If I have to write `with_preset(WasdKeys)`, I'm thinking, "what the hell is a preset? I just want to `bind` `Move` to `WasdKeys`." -> `ctx.bind::<Move>().with(WasdKeys)`

Re your game code, this is a great example of DX we can improve. We should do 2 things:
- `impl BindPreset for Vec<InputBind>`
  - more generally, `for impl IntoIterator<impl Into<InputBind>>`
- `XyAxes<Up, Left, Down, Right>` where each type parameter is itself an `impl BindPreset`

Then we can turn this code:
```rs
        for &key in &settings.keyboard.camera_forward {
            move_action.with(InputBind::new(key).with_modifier(SwizzleAxis::YXZ));
        }
        for &key in &settings.keyboard.camera_left {
            move_action.with(InputBind::new(key).with_modifier(Negate::default()));
        }
        for &key in &settings.keyboard.camera_backward {
            move_action.with(
                InputBind::new(key)
                    .with_modifier(SwizzleAxis::YXZ)
                    .with_modifier(Negate::default()),
            );
        }
        for &key in &settings.keyboard.camera_right {
            move_action.with(key);
        }
```
into
```rs
move_action.with(XyAxes::new(
    &settings.keyboard.camera_forward,
    &settings.keyboard.camera_left,
    &settings.keyboard.camera_backward,
    &settings.keyboard.camera_right,
});
```

Also, there should be an example which uses code similar to your game's control settings. It demonstrates a great pattern which would be commonly used by others.

Reply:
Hm... Not sure about `with` vs `with_preset... I think that the distinction between inputs and presets are quite logical. It just creates another level of structs indirection that we pass to this function, but without much practical benefit. Let me ask one more person opinion about it.

Agree with the `XyAxes` suggestion!

Reply:
> north/south/east/west? I

Agree!